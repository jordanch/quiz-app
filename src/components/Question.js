import React from "react"
import Typography from "@material-ui/core/Typography"
import { withStyles } from "@material-ui/core/styles"
import Button from "@material-ui/core/Button"
import PropTypes from "prop-types"

const styles = (theme) => ({
	container: {
		width: "100%",
		height: "100%",
		backgroundColor: "white",
		display: "flex",
		flexDirection: "column",
		alignItems: "center",
		justifyContent: "space-around"
	},
	questionBox: {
		border: "1px solid black",
		width: "80%",
		padding: "20px 10px"
	},
	answerContainer: {
		display: "flex",
		alignItems: "center",
		justifyContent: "space-around"
	},
	button: {
		margin: theme.spacing.unit
	}
})

const Question = (props) => {
	const {
		classes,
		quizEntities,
		currentQuizEntityIndex,
		answerQuizEntity,
		cbAfterLastQuestion
	} = props

	// TODO: is this an efficient way of transitioning to /score route when last question answered?
	// I'd like to get this logic out of this component. it has to know about /score and logic for last question
	const quizEntitiesLength = quizEntities.length
	const current = quizEntities[currentQuizEntityIndex]
	const { category, question } = current

	function shouldCallLastQuestionCb(currentIndex) {
		if (currentIndex + 1 === quizEntitiesLength) {
			cbAfterLastQuestion()
		}
	}

	return (
		<div className={classes.container}>
			<Typography variant="headline" gutterBottom align="center">
				{category}
			</Typography>

			<div className={classes.questionBox}>
				<Typography variant="body2" gutterBottom align="center">
					{question}
				</Typography>
			</div>

			<div className={classes.answerContainer}>
				<Button
					variant="contained"
					color="secondary"
					className={classes.button}
					onClick={() => {
						answerQuizEntity(currentQuizEntityIndex, false)
						shouldCallLastQuestionCb(currentQuizEntityIndex)
					}}
				>
					FALSE
				</Button>
				<Button
					variant="contained"
					color="primary"
					className={classes.button}
					onClick={() => {
						answerQuizEntity(currentQuizEntityIndex, true)
						shouldCallLastQuestionCb(currentQuizEntityIndex)
					}}
				>
					TRUE
				</Button>
			</div>

			<Typography variant="body2" gutterBottom align="center">
				{`${currentQuizEntityIndex + 1} of ${quizEntitiesLength}`}
			</Typography>
		</div>
	)
}

Question.propTypes = {
	classes: PropTypes.object.isRequired,
	// TODO: add shape of quizEntity.
	quizEntities: PropTypes.array.isRequired,
	currentQuizEntityIndex: PropTypes.number.isRequired,
	answerQuizEntity: PropTypes.func.isRequired,
	cbAfterLastQuestion: PropTypes.func.isRequired
}

export default withStyles(styles)(Question)
