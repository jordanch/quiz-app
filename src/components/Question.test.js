import React from 'react';
import Question from './Question';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Question/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});