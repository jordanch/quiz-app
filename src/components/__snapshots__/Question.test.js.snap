// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Question-container-1"
>
  <h1
    className="MuiTypography-root-5 MuiTypography-headline-10 MuiTypography-gutterBottom-22 MuiTypography-alignCenter-18"
  >
    Category
  </h1>
  <div
    className="Question-questionBox-2"
  >
    <aside
      className="MuiTypography-root-5 MuiTypography-body2-13 MuiTypography-gutterBottom-22 MuiTypography-alignCenter-18"
    >
      Question here... what is the queen of England's shoe size?
    </aside>
  </div>
  <div
    className="Question-answerContainer-3"
  >
    <button
      className="MuiButtonBase-root-56 MuiButton-root-30 MuiButton-contained-41 MuiButton-containedSecondary-43 MuiButton-raised-44 MuiButton-raisedSecondary-46 Question-button-4"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-31"
      >
        FALSE
      </span>
      <span
        className="MuiTouchRipple-root-59"
      />
    </button>
    <button
      className="MuiButtonBase-root-56 MuiButton-root-30 MuiButton-contained-41 MuiButton-containedPrimary-42 MuiButton-raised-44 MuiButton-raisedPrimary-45 Question-button-4"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-31"
      >
        TRUE
      </span>
      <span
        className="MuiTouchRipple-root-59"
      />
    </button>
  </div>
  <aside
    className="MuiTypography-root-5 MuiTypography-body2-13 MuiTypography-gutterBottom-22 MuiTypography-alignCenter-18"
  >
    1 of 10
  </aside>
</div>
`;
